src/routes/safe/store/actions/transactions/utils/transactionHelpers.ts

```diff
@@ -0,0 +1,308 @@
+  const { decimals = null, symbol = null } = isSendERC20Tx ? await getERC20DecimalsAndSymbol(tx.to) : {}
+
+  const txToStore = makeTransaction({
+    baseGas: tx.baseGas,
+    blockNumber: tx.blockNumber,
+    cancelled: isTxCancelled,
+    confirmations,
+    creationTx: tx.creationTx,
+    customTx: isCustomTx,
+    data: tx.data ? tx.data : EMPTY_DATA,
+    decimals,
+    decodedParams,
+    executionDate: tx.executionDate,
+    executionTxHash: tx.transactionHash,
+    executor: tx.executor,
+    gasPrice: tx.gasPrice,
+    gasToken: tx.gasToken || ZERO_ADDRESS,
+    isCancellationTx,
+    isCollectibleTransfer: isSendERC721Tx,
+    isExecuted: tx.isExecuted,
+    isSuccessful: tx.isSuccessful,
+    isTokenTransfer: isSendERC20Tx,
+    modifySettingsTx: isModifySettingsTx,
+    multiSendTx: isMultiSendTx,
+    nonce: tx.nonce,
+    operation: tx.operation,
+    origin: tx.origin,
+    recipient: tx.to,
+    refundParams,
+    refundReceiver: tx.refundReceiver || ZERO_ADDRESS,
+    safeTxGas: tx.safeTxGas,
+    safeTxHash: tx.safeTxHash,
+    submissionDate: tx.submissionDate,
+    symbol,
+    upgradeTx: isUpgradeTx,
+    value: tx.value.toString(),
+  })
+
+  return txToStore
+    .set('status', calculateTransactionStatus(txToStore, safe, currentUser))
+    .set('type', calculateTransactionType(txToStore))
+}
+
+export const mockTransaction = (tx, safeAddress: string, state): Promise<any> => {
+  const submissionDate = new Date().toISOString()
+
+  const transactionStructure: any = {
+    blockNumber: null,
+    confirmationsRequired: null,
+    dataDecoded: decodeMethods(tx.data),
+    ethGasPrice: null,
+    executionDate: null,
+    executor: null,
+    fee: null,
+    gasUsed: null,
+    isExecuted: false,
+    isSuccessful: null,
+    modified: submissionDate,
+    origin: null,
+    safe: safeAddress,
+    safeTxHash: null,
+    signatures: null,
+    submissionDate,
+    transactionHash: null,
+    confirmations: [],
+    ...tx,
+  }
+
+  const knownTokens = state[TOKEN_REDUCER_ID]
+  const safe = state[SAFE_REDUCER_ID].getIn([SAFE_REDUCER_ID, safeAddress])
+  const cancellationTxs = state[CANCELLATION_TRANSACTIONS_REDUCER_ID].get(safeAddress) || Map()
+  const outgoingTxs = state[TRANSACTIONS_REDUCER_ID].get(safeAddress) || List()
+
+  return buildTx({
+    cancellationTxs,
+    currentUser: null,
+    knownTokens,
+    outgoingTxs,
+    safe,
+    tx: transactionStructure,
+    txCode: EMPTY_DATA,
+  })
+}
+
+export const updateStoredTransactionsStatus = (dispatch, walletRecord) => {
+  const state = store.getState()
+  const safe = safeSelector(state)
+  if (safe) {
+    const safeAddress = safe.address
+    const transactions = safeTransactionsSelector(state)
+    dispatch(
+      addOrUpdateTransactions({
+        safeAddress,
+        transactions: transactions.withMutations((list) =>
+          list.map((tx) => tx.set('status', calculateTransactionStatus(tx, safe, walletRecord.account))),
```

Comment:
As I said in one of other comments I think we need to make a separate component and selector to calculate the status from the state, because IMO it would be easier for persisting data, no need to update two pieces of data (pending actions and status)